data/mp/multiplay/maps/6c-Entropy/game.js

```diff
@@ -482,21 +490,43 @@ function placeStuff(regions, startPos) {
 			structures.push({name: "A0ResourceExtractor", position: oilPos, direction: 0x4000*gameRand(4), modules: 0, player: player});
 			structures.push({name: "WallTower01", position: placeNear(oilPos[0]/128, oilPos[1]/128, 1, 1, false, 4), direction: 0x4000*gameRand(4), modules: 0, player: player});
 		}
-		droids.push({name: "ConstructionDroid", position: placeNear(x, y, 1, 1, false, 4), direction: gameRand(0x10000), player: player});
-		droids.push({name: "ConstructionDroid", position: placeNear(x, y, 1, 1, false, 4), direction: gameRand(0x10000), player: player});
+		for(var i=0;i<trucksPerPlayer;i++){
+			droids.push({name: "ConstructionDroid", position: placeNear(x, y, 1, 1, false, 4), direction: gameRand(0x10000), player: player});
+		}
 	}
 	if (placeNearFailed) {
 		if (verbose) log('Base placement failed!');
 		return null;  // Failed to place something important.
 	}
-	var featureTypes = ["Tree1", "Tree2", "Tree3", "Tree1", "Tree2", "Tree3", "LogCabin1", "LogCabin2", "WaterTower"];
-	var snowFeatureTypes = ["TreeSnow1", "TreeSnow2", "TreeSnow3", "TreeSnow1", "TreeSnow2", "TreeSnow3", "LogCabin1", "LogCabin2", "WaterTower"];
-	for (var i = 0; i < 150; ++i) {
-		var pos = placeNear(gameRand(mapWidth), gameRand(mapHeight), 1, 1, true);
-		if (pos !== null) {
-			var x = pos[0]/128 | 0, y = pos[1]/128 | 0;
-			var snow = isSnow[mapWidth*y + x];
-			features.push({name: sample(snow? snowFeatureTypes : featureTypes), position: pos, direction: gameRand(0x10000)});
+	if(decorationMode!=0){
+		var featureTypes = ["Tree1", "Tree2", "Tree3", "Tree1", "Tree2", "Tree3", "LogCabin1", "LogCabin2", "WaterTower"];
+		var snowFeatureTypes = ["TreeSnow1", "TreeSnow2", "TreeSnow3", "TreeSnow1", "TreeSnow2", "TreeSnow3", "LogCabin1", "LogCabin2", "WaterTower"];
+		for (var i = 0; i < mapSize*decorationDensity; ++i) {
+			var pos = placeNear(gameRand(mapWidth), gameRand(mapHeight), 1, 1, true);
+			if (pos !== null) {
```

Comment:
I think using

    if (pos === null) {
        continue;
    }
    ...

... would make the code a bit more easy to read, by avoiding this much levels of indentation. Same for the other ifs below.